web_src/js/features/pull-view-file.js

```diff
@@ -53,9 +53,17 @@ export function initViewedCheckboxListenerFor() {
       const hasChangedLabel = form.parentNode.querySelector('.changed-since-last-review');
       hasChangedLabel?.remove();
 
+      const fileName = checkbox.getAttribute('name');
+
+      // check if the file is in our diffFileInfo and if we find it -> change the IsViewed status
+      const fileInPageData = pageData.diffFileInfo?.files?.find((x) => x.Name === fileName);
```

Comment:
not sure if this is the best way to sync the status - any feedback is much appreciated

Reply:
I think you could sync the state between the components through a store instead:

https://github.com/go-gitea/gitea/blob/main/web_src/js/modules/stores.js

Reply:
Alternatively, we could also extract the file tree updating logic into a separate method, which is probably even the best course of action regarding maintainability.

Reply:
It needs to use "event" or "vue reactive store" to pass the values. Store is the best choice.

Otherwise the reactive system would break sooner or later (it had been broken before)

Reply:
thanks @wxiaoguang - i've now tried to put everything into the store. 
Accessing the store from box.tmpl was not that easy - could you please look at the solution and provide feedback?